include/aws/auth/private/credentials_utils.h

```diff
@@ -166,15 +166,22 @@ AWS_AUTH_API
 enum aws_retry_error_type aws_credentials_provider_compute_retry_error_type(int response_code, int error_code);
 
 /*
- * Constructs an endpoint in the format of service_name.region.amazonaws.com
+ * If an endpoint override is configured, use that. Otherwise do the following:
+ * Construct an endpoint in the format of service_name_host.region.amazonaws.com.
  * If the region is cn-north-1 or cn-northwest-1, .cn is appended to support China-specific regional endpoints.
+ * To configure the endpoint override, check the
+ * https://docs.aws.amazon.com/sdkref/latest/guide/feature-ss-endpoints.html.
  */
 AWS_AUTH_API
-int aws_credentials_provider_construct_regional_endpoint(
+int aws_credentials_provider_construct_endpoint(
```

Comment:
I'm not proposing an options-struct because this is private

But if someone moves this out of private/ and is like "WHY SHOULD I CHANGE IT YOU ALREADY APPROVED THE EXISTING CODE"

then I can point at this comment here and say "NUH UH"